packages/cli/test/test-utils.js

```diff
@@ -86,8 +87,16 @@ async function startServer(sqlFile, extraArgs = []) {
     ...extraArgs,
   ]);
   serverProcess.stdout.on('data', chunk => (stdout += chunk.toString()));
+  serverProcess.stderr.on('data', chunk => (stderr += chunk.toString()));
+
+ 
+  const unexpectedServerClose = code => {
+    throw new Error(`Server process closed unexpectedly.\n\n${stderr}`);
+  };
+  serverProcess.on('close', unexpectedServerClose);
 
   await waitForCondition(() => stdout.includes('listening'));
+  serverProcess.removeListener('close', unexpectedServerClose);
```

Comment:
```suggestion
  try {
    await waitForCondition(() => stdout.includes('listening'));
  } catch (err) {
    throw new Error(`Server failed to start: ${err.message}\n\n${stdout}\n\n${stderr}`);
  }
```

how do you feel about making it more generic for *any* failure?